docs/source/export.ir_spec.rst

```diff
@@ -0,0 +1,501 @@
+  the graph, they are lifted as inputs to the graph. The graph_signature is
+  utilized to keep track of additional information on these parameters and
+  buffers.
+- ``state_dict`` (``Dict[str, Union[torch.Tensor, torch.nn.Parameter]]``): Data
+  structure containing the parameters and buffers.
+- ``range_constraints`` (``Dict[sympy.Symbol, RangeConstraint]``): For programs
+  that are exported with data dependent behavior, the metadata on each node will
+  contain symbolic shapes (which look like ``s0``, ``i0``). This attribute maps
+  the symbolic shapes to their lower/upper ranges.
+- ``equality_constraints`` (``List[Tuple[InputDim, InputDim]]``): A list of
+  nodes in the graph and dimensions that have the same shape.
+
+Graph
+-----
+
+An Export IR Graph is a PyTorch program represented in the form of a DAG
+(directed acyclic graph).  Each node in this graph represents a particular
+computation or operation, and edges of this graph consist of references between
+nodes.
+
+We can view Graph having this schema:
+
+.. code-block:: python
+
+   class Graph:
+     nodes: List[Node]
+
+In practice, Export IR's graph is realized as :class:`torch.fx.Graph` Python class.
+
+An Export IR graph contains the following nodes (Nodes will be described in more
+details in the next section):
+
+- 0 or more nodes of op type ``placeholder``
+- 0 or more nodes of op type ``call_function``
+- exactly 1 node of op type ``output``
+
+**Collorary:** The smallest valid Graph will be of one node. i.e. nodes is never empty.
+
+**Definition:**
+The set of ``placeholder`` nodes of a Graph represents the **inputs** of the
+Graph of GraphModule. The `output` node of a Graph represents the **outputs**
+of the Graph of GraphModule.
+
+Example::
+
+   from torch import nn
+
+   class MyModule(nn.Module):
+
+       def forward(self, x, y):
+         return x + y
+
+   mod = torch._export.export(MyModule())
+   print(mod.graph)
+
+The above is the textual representation of a Graph, with each line being a node.
+
+Node
+----
+
+A Node represents a particular computation or operation and is represented in
+Python using the :class:`torch.fx.Node` class. Edges between nodes are
+represented as direct references to other nodes via the ``args`` property of the
+Node class. Using the same FX machinery, we can represent the following
+operations that a computational graph typically needs, such as operator calls,
+placeholders (aka inputs), conditionals, and loops.
+
+The Node has the following schema:
+
+.. code-block:: python
+
+   class Node:
+     name: str # name of node
+     op_name: str  # type of operation
+
+     # interpretation of the fields below depends on op_name
+     target: [str|Callable]
+     args: List[object]
+     kwargs: Dict[str, object]
+     meta: Dict[str, object]
+
+**FX Text Format**
+
+As in the example above, notice that each line has this format::
+
+   %<name>:[...] = <op_name>[target=<target>](args = (%arg1, %arg2, arg3, arg4, …)), kwargs = {})
```

Comment:
Is this still accurate? I think a while ago we were talking about supporting kwargs?